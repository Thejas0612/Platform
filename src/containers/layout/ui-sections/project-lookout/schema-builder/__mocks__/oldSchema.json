[
  {
    "componentName": "DynamicForm",
    "alignment": "right",
    "componentProps": {
      "schema": [
        {
          "id": 0,
          "fields": [
            {
              "column": 12,
              "label": "LABEL_TEXT",
              "labelClass": "app-content-label",
              "name": "LABEL_TEXT",
              "showLabel": false,
              "subText": "",
              "tableClass": "",
              "text": "Measurement Type",
              "textClass": "ddl-typography--h5",
              "title": "this is title",
              "type": "LABEL_TEXT",
              "textStyles": {
                "textAlign": "start"
              }
            },
            {
              "column": 12,
              "type": "TILE_THUMBNAIL",
              "name": "measurement-type",
              "defaultIds": [],
              "data": [
                {
                  "id": "FLOW",
                  "title": "Flow",
                  "description": "Flow Measurement is the process of measuring fluid in your plant or industry. You can measure flow through a variety of different devices such as Coriolis, differential pressure, vortex, magnetic, ultrasonic, turbine and positive displacement meters.",
                  "imgUrl": "https://emerson-cdn.azurewebsites.net/7bd555544cf68071bafa.png"
                },
                {
                  "id": "DENSITY",
                  "title": "Density",
                  "description": "Our density measurement devices offer reliable performance for applications such as concentration, API, Brix, and more.",
                  "imgUrl": "https://emerson-cdn.azurewebsites.net/940becae8b116d587fbd.png"
                },
                {
                  "id": "VISCOSITY",
                  "title": "Viscosity",
                  "description": "Our inline viscosity meter offers accurate, repeatable measurement in industries such as refining, chemical, life sciences, and more.",
                  "imgUrl": "https://emerson-cdn.azurewebsites.net/940becae8b116d587fbd.png"
                }
              ],
              "required": true
            }
          ]
        }
      ]
    }
  }
]