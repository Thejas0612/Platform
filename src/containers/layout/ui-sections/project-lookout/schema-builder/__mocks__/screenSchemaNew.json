[
  {
    "id": 0,
    "fields": [
      {
        "column": 12,
        "label": "LABEL_TEXT",
        "labelClass": "app-content-label",
        "name": "LABEL_TEXT",
        "showLabel": false,
        "subText": "",
        "tableClass": "",
        "text": "Measurement Type",
        "textClass": "ddl-typography--h5",
        "title": "this is title",
        "type": "LABEL_TEXT",
        "textStyles": {
          "textAlign": "start"
        }
      },
      {
        "column": 12,
        "type": "TILE_THUMBNAIL",
        "name": "measurement-type",
        "defaultIds": [
          "FLOW"
        ],
        "data": [
          {
            "id": "FLOW",
            "title": "Flow",
            "description": "Flow Measurement is the process of measuring fluid in your plant or industry. You can measure flow through a variety of different devices such as Coriolis, differential pressure, vortex, magnetic, ultrasonic, turbine and positive displacement meters.",
            "imgUrl": "https://emerson-cdn.azurewebsites.net/7bd555544cf68071bafa.png",
            "disabled": false
          },
          {
            "id": "DENSITY",
            "title": "Density",
            "description": "Our density measurement devices offer reliable performance for applications such as concentration, API, Brix, and more.",
            "imgUrl": "https://emerson-cdn.azurewebsites.net/940becae8b116d587fbd.png"
          },
          {
            "id": "VISCOSITY",
            "title": "Viscosity",
            "description": "Our inline viscosity meter offers accurate, repeatable measurement in industries such as refining, chemical, life sciences, and more.",
            "imgUrl": "https://emerson-cdn.azurewebsites.net/940becae8b116d587fbd.png",
            "disabled": true,
            "tooltip": "There are no products that measure both Flow and Viscosity."
          }
        ],
        "required": true,
        "value": [
          "FLOW"
        ]
      },
      {
        "column": 12,
        "name": "process-piping",
        "text": "Process Piping",
        "type": "LABEL_TEXT",
        "value": [
          "FLOW"
        ]
      }
    ]
  },
  {
    "id": 2,
    "fields": [
      {
        "type": "TABLE_INPUT",
        "column": 12,
        "name": "TABLE_INPUT2",
        "data": [
          [
            {
              "type": "TEXT",
              "label": "",
              "name": "r1height",
              "align": "center"
            },
            {
              "type": "TEXT",
              "label": "Min",
              "align": "center"
            },
            {
              "type": "TEXT",
              "label": "Normal",
              "align": "center"
            },
            {
              "type": "TEXT",
              "label": "Max",
              "align": "center"
            },
            {
              "type": "TEXT",
              "label": "Units",
              "align": "center"
            }
          ],
          [
            {
              "type": "SINGLE_SELECT",
              "name": "density_options",
              "value": "",
              "disabled": true,
              "options": [
                {
                  "value": "Density",
                  "label": "Density",
                  "selected": true
                },
                {
                  "value": "specific_gravity",
                  "label": "Specific Gravity"
                }
              ]
            },
            {
              "type": "TEXT_INPUT",
              "name": "density_min",
              "disabled": true,
              "align": "center",
              "precision": 4,
              "min": 0.0001,
              "minError": "Entered Minimum Density is below or equal to 0"
            },
            {
              "type": "TEXT_INPUT",
              "name": "density_norm",
              "inputClass": "customRequired",
              "disabled": true,
              "required": false,
              "align": "center",
              "precision": 4,
              "min": 0.0001,
              "minError": "Entered Normal Density is below or equal to 0"
            },
            {
              "type": "TEXT_INPUT",
              "name": "density_max",
              "disabled": true,
              "align": "center",
              "precision": 4,
              "min": 0.0001,
              "minError": "Entered Maximum Density is below or equal to 0"
            },
            {
              "type": "SINGLE_SELECT",
              "name": "density_unit",
              "value": "Meter",
              "disabled": false,
              "align": "center",
              "inputClass": "unitClass",
              "options": [
                {
                  "value": "kg/m3",
                  "label": "kg/m3"
                },
                {
                  "value": "g/cm3",
                  "label": "g/cm3",
                  "selected": true
                },
                {
                  "value": "lb/ft3",
                  "label": "lb/ft3"
                },
                {
                  "value": "lb/gallon(US)",
                  "label": "lb/gallon(US)"
                }
              ]
            }
          ]
        ],
        "title": "",
        "tableClass": "",
        "labelClass": "app-content-label"
      }
    ]
  }
]
