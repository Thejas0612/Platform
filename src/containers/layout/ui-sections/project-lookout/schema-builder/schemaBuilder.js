import { WorkflowBuilder } from "./WorkflowBuilder";

/**
 * Updates schema using fluent interface design pattern.
 * @param oldSchema {object[]}
 * @return {WorkflowBuilder}
 */
export function schemaBuilder(oldSchema) {
  return new WorkflowBuilder(oldSchema);
}
