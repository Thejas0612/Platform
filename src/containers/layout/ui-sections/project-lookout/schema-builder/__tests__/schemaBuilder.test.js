import oldSchema from "../__mocks__/oldSchema.json";
import screenSchemaNew from "../__mocks__/screenSchemaNew.json";
import { schemaBuilder } from "../schemaBuilder";

const VISCOSITY_ID = "VISCOSITY";

describe("schemaBuilder", () => {
  it("when measurement type is update, then disable ", () => {
    const schema = schemaBuilder(oldSchema)
      .screen(0)
      .tileThumbnail("measurement-type")
      .onChange((field) => {
        const viscosityOption = field.data?.find(_ => _.id === VISCOSITY_ID);
        if (viscosityOption == null) {
          throw new Error("Could not find viscosityOption.");
        }

        viscosityOption.disabled = true
      })
      .build(0, screenSchemaNew);

    expect(schema).toMatchSnapshot();
  });
});