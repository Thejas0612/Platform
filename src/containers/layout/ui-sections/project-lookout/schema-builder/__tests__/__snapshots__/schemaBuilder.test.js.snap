// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schemaBuilder when measurement type is update, then disable  1`] = `
[
  {
    "alignment": "right",
    "componentName": "DynamicForm",
    "componentProps": {
      "schema": [
        {
          "fields": [
            {
              "column": 12,
              "label": "LABEL_TEXT",
              "labelClass": "app-content-label",
              "name": "LABEL_TEXT",
              "showLabel": false,
              "subText": "",
              "tableClass": "",
              "text": "Measurement Type",
              "textClass": "ddl-typography--h5",
              "textStyles": {
                "textAlign": "start",
              },
              "title": "this is title",
              "type": "LABEL_TEXT",
            },
            {
              "column": 12,
              "data": [
                {
                  "description": "Flow Measurement is the process of measuring fluid in your plant or industry. You can measure flow through a variety of different devices such as Coriolis, differential pressure, vortex, magnetic, ultrasonic, turbine and positive displacement meters.",
                  "disabled": false,
                  "id": "FLOW",
                  "imgUrl": "https://emerson-cdn.azurewebsites.net/7bd555544cf68071bafa.png",
                  "title": "Flow",
                },
                {
                  "description": "Our density measurement devices offer reliable performance for applications such as concentration, API, Brix, and more.",
                  "id": "DENSITY",
                  "imgUrl": "https://emerson-cdn.azurewebsites.net/940becae8b116d587fbd.png",
                  "title": "Density",
                },
                {
                  "description": "Our inline viscosity meter offers accurate, repeatable measurement in industries such as refining, chemical, life sciences, and more.",
                  "disabled": true,
                  "id": "VISCOSITY",
                  "imgUrl": "https://emerson-cdn.azurewebsites.net/940becae8b116d587fbd.png",
                  "title": "Viscosity",
                  "tooltip": "There are no products that measure both Flow and Viscosity.",
                },
              ],
              "defaultIds": [
                "FLOW",
              ],
              "name": "measurement-type",
              "required": true,
              "type": "TILE_THUMBNAIL",
              "value": [
                "FLOW",
              ],
            },
          ],
          "id": 0,
        },
      ],
    },
  },
]
`;
